### PR TITLE
Feature: Add verbose message informing which dependencies were skipped

### LINKED ISSUE #15: Add verbose messages informing which dependencies were skipped
Since we have flags `--production` and `--no-optional` we should show which dependencies are we skipping while analyzing the dependencies.

Something like this:

For `--production`:

```
Analyzing 15 dependencies (skipping devDependencies) ...
```

For `--no-optional`:

```
Analyzing 15 dependencies (skipping optionalDependencies) ...
```

For `--no-optional --production`:

```
Analyzing 15 dependencies (skipping devDependencies and optionalDependencies) ...
```

I have a fix ready for this. I'll push once #14 is merged.

/cc @nolanlawson

### PR DESCRIPTION
**Pull request by sahildua2305**
Fixes #15 

/cc @nolanlawson

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -32,14 +32,31 @@ function getDeps () {
   return readFile('package.json', 'utf8').then(function (str) {
     var json = JSON.parse(str)
     var deps = extend({}, json.dependencies)
+    var skipped = {
+      dev: 0,
+      optional: 0
+    }
```
Comment (nolanlawson) [inline index.js:38]: I would avoid creating an object here and just use two `var`s.

File: index.js [hunk 2]
```diff
@@ -32,14 +32,31 @@ function getDeps () {
...
     if (process.argv.indexOf('--production') === -1) {
       extend(deps, json.devDependencies)
+    } else {
+      skipped.dev = json.devDependencies ? Object.keys(json.devDependencies).length : 0
```
Comment (nolanlawson) [inline index.js:42]: `Object.keys(json.devDependencies || {}).length` is a bit terser.

File: index.js [hunk 3]
```diff
@@ -32,14 +32,31 @@ function getDeps () {
...
     }
     // Include optionalDependencies only if --no-optional argument is absent
     if (process.argv.indexOf('--no-optional') === -1) {
       extend(deps, json.optionalDependencies)
+    } else {
+      skipped.optional = json.optionalDependencies ? Object.keys(json.optionalDependencies).length : 0
```
Comment (nolanlawson) [inline index.js:48]: Ditto ^

File: index.js [hunk 4]
```diff
@@ -32,14 +32,31 @@ function getDeps () {
...
+    }
+    var analysis_start_msg = 'Analyzing ' + Object.keys(deps).length + ' dependencies'
```
Comment (nolanlawson) [inline index.js:50]: JavaScript typically uses `lowerCamelCase`, `snake_case_like_this` is more typical for Python.

**Comment by nolanlawson** [comment]
Besides a few nitpicks, looks good to me!

**Comment by nolanlawson** [comment]
Merged as 6cb30ede1b40288b74dfb94b8ee0a8c313b8e5cb, fixed nitpicks in 6e09741d91afaa1e9453f3175f2bd83a4a594d78 and fd5c89d151a570eb2314f7ddf7e8d37e1b849fd1, thank you!

**Comment by nolanlawson** [comment]
All features published in 1.3.0

**Comment by sahildua2305** [comment]
Thanks @nolanlawson :smile:

**Pull request closed without merging**